9 subissue 11 backend skeleton

MMachadoUofT opened the pull request:
Considering this is my pull request, let me know what you want to see next time. Thankfully this is pretty bare-bones, just creating the entity interfaces so people can write their code without having IntelliJ scream at them.

MMachado05 committed:
Created source files for the backend entities

This includes the Curriculum, Task, TaskTree, TimeBlock, TimeBlockManager, and User, as well as the interfaces they implement for the sake of abstraction.

MMachado05 committed:
Tentatively completed the skeletons for each of the classes in the backend

In particular, the interfaces for each of the classes have been completed unless otherwise decided, and the User and Schedule classes have been added as well. None of the actual implemented classes work yet; these will be dealt with individually through their respective issues. As far as my work goes, I won't be touching the User or Schedule classes anymore unless otherwise decided.

bmguiler commented on src/main/java/entityLayer/CommonCurriculum.java:
```diff
@@ -1,4 +1,20 @@
+    public TimeBlockManager getThisWeekSchedule() {
+        return null;
+    }
 }
```
also a method getName so we know which curriculum we are dealing with within the schedule

MMachadoUofT replied:
Got it!

bmguiler commented on src/main/java/entityLayer/CommonSchedule.java:
```diff
@@ -0,0 +1,18 @@
+    public Curriculum getCurriculum(int curriculumID) {
+        return null;
+    }
+}
```
method for changeAvailability

MMachadoUofT replied:
I'll go ahead and put that in, but I'm going to hold off a while on actually implementing. We've got 10 days until the deadline as of me writing this reply, and creating a "changeavailability" method will require also changing every single curriculum's TimeManager objects to fit into the new availability, which will probably also have the user have to review it, etc, etc.While I'd love to implement that functionality, it just isn't as pressing as our more foundational functionalities, and poses way too many complications for us to begin considering implementing it this early into the process.

I'll type it into the interface just to have it, but I'll make a note in the documentation that it has yet to be implemented, and that the implementation will be something to be decided on later.

bmguiler commented on src/main/java/entityLayer/CommonTimeBlockManager.java:
```diff
@@ -1,4 +1,20 @@
+    public boolean removeTimeBlock(TimeBlock timeBlock) {
+        return false;
+    }
 }
```
we talked about a method for checking to make sure that a time block lies within the users availability, not sure if that will still be useful or not

MMachadoUofT replied:
The TimeBlocks have methods to see whether they are contained within/overlap with other TimeBlocks, so I suspected that we'd simply be using those instead. As I last remember from how we discussed it, the availability is a manager with only 7 timeblocks (one for each day of the week), so doing this shouldn't be too difficult, I don't think.

bmguiler commented on src/main/java/entityLayer/CommonUser.java:
```diff
@@ -1,4 +1,23 @@
+    public void setPassword() {
+
+    }
 }
```
will the user have a username too or just an email?

bmguiler replied:
oh also a method for getAvailability

MMachadoUofT replied:
I'm looking into adding all of these now; However, I'm gonna hold off on creating a getAvailability method for the user, instead having client code first get the _schedule_, then calling the Schedule's getAvailability method (which, looking now, I'm having to implement. Oops. Guess that's why we do code reviews!)

bmguiler reviewed (dismissed):
Hey Marcial this is really great. I know you said that this is bare bones just to get us started which I really appreciate, I just added a few comments of some other methods that we talked about implementing in some of the classes

dadler83 reviewed (changes requested):
For all of the classes there should be file doc strings and method doc strings, and for all of the interfaces there should be file docstrings that reference the classes they interact with.

Also, you should add your name as an author to all of the file doc strings along with a date created, and last updated. This lets us more easily track who has worked on what, the function of every class / interface, and when these changes were made.

MMachado05 committed:
Added documentation for the User interface

I also added some username-related methods, as well as a method for returning whether the user's current email was valid in format.

MMachado05 committed:
Added documentation for the CommonUser class

This mostly involved changing the auto-generated interface documentation to say "CommonUser" as opposed to just "User"

MMachado05 committed:
Added documentation for the Schedule interface

Just like the last one, this was just making JavaDocs, really. That, and writing a getAvailability() method that I somehow forgot to include.

MMachado05 committed:
Added documentation for the Schedule class

This was mostly just the class documentation; much of the implemented methods remained the same.

MMachado05 committed:
Added documentation for the Task interface.

Methods and the class' creation info.

MMachado05 committed:
Added documentation for the CommonTask class.

Methods and the class' creation info.

MMachado05 committed:
Added documentation for the Curriculum interface.

Methods and the interface's creation info.

MMachado05 committed:
Added documentation for the CommonCurriculum class.

Methods and the class's creation info.

MMachado05 committed:
Added documentation for the TaskTree interface.

I've added a few more methods as well that ought to prove useful in implementing this class' functionality; it's a pretty dense one where a lot of the magic actually happens.

MMachado05 committed:
Added documentation for the CommonTaskTree class

MMachado05 committed:
Added documentation for the TimeBlock interface.

I've also created a few more functionality methods for comparing TimeBlocks to one another.

MMachado05 committed:
Added documentation for the CommonTimeBlock class.

MMachado05 committed:
Added documentation for the TimeBlockManager interface.

MMachado05 committed:
Added documentation for the CommonTimeBlockManager class.

od-obas1187 commented:
Looks good so far!

bmguiler reviewed (commented):
looks really great thanks for getting this done so quickly
